lib/puppet/parser/functions/docker_run_flags.rb

```diff
@@ -5,17 +5,27 @@
 # docker_run_flags.rb
 #
 module Puppet::Parser::Functions
+  newfunction(:'docker::escape', type: :rvalue) do |args|
+    subject = args[0]
+
+    if self['facts'] && self['facts']['os']['family'] == 'windows'
+      %("#{subject.gsub('"', '\\"')}")
```

Comment:
is that just because the shellescape function is designed for linux? what about patching https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/lib/puppet/parser/functions/shell_escape.rb instead?

Reply:
The more I am thinking about it, the more I am concerned about cmd.exe escaping vs. powershell escaping: they seem to be different in incompatible ways:

* https://ss64.com/nt/syntax-esc.html
* https://ss64.com/ps/syntax-esc.html

Maybe the stdlib should have 3 shell escape functions: `shell_escape`, `powershell_escape` and `cmd_escape`, then modules could have a local escape function which call the right escape function on windows depending on what the module is using?

Reply:
that sounds like a good idea

Reply:
https://github.com/puppetlabs/puppetlabs-stdlib/pull/1235 was merged in stdlib. When we have a new release with this code, we will be able to move on with this PR.

Reply:
Awesome! I added a commit to rely on this new code.  Let's see what the CI think about it :wink: